include/AST/Types/StructTy.hpp

```diff
@@ -0,0 +1,67 @@
+#ifndef GLU_AST_TYPES_STRUCTTY_HPP
+#define GLU_AST_TYPES_STRUCTTY_HPP
+
+#include "Basic/SourceLocation.hpp"
+#include "TypeBase.hpp"
+
+#include <llvm/ADT/SmallVector.h>
+#include <llvm/ADT/StringRef.h>
+#include <llvm/AST/STLExtras.h>
+#include <string>
+
+namespace glu::types {
+
+/// @brief StructTy is a class that represents structures declared in code.
+class StructTy : public TypeBase {
+
+    struct Field {
+        std::string name;
+        TypeBase *type;
+    };
+
+    std::string _name;
+    llvm::SmallVector<Field> _fields;
+    SourceLocation _definitionLocation;
+
+    // TODO: Add attributes (e.g. packed, alignment)
+
+public:
+    /// @brief Constructor for the StructTy class.
+    StructTy(
+        std::string name, llvm::SmallVector<Field> fields,
+        SourceLocation definitionLocation
+    )
+        : TypeBase(TypeKind::StructTyKind)
+        , _name(name)
+        , _fields(fields)
+        , _definitionLocation(definitionLocation)
+    {
+    }
+
+    std::string getName() const { return _name; }
+
+    size_t getFieldCount() const { return _fields.size(); }
+    Field const &getField(size_t index) const { return _fields[index]; }
```

Comment:
assert for index ?

Reply:
the assertion is already done inside of `llvm::SmallVector::operator[]` but we can add one here if you want?